schematic/store/synapse.py

```diff
@@ -1724,12 +1720,19 @@ def getDatasetProject(self, datasetId: str) -> str:
             str: The Synapse ID for the parent project.
         """
 
-        self._query_fileview()
-
         # Subset main file view
         dataset_index = self.storageFileviewTable["id"] == datasetId
         dataset_row = self.storageFileviewTable[dataset_index]
 
+        # re-query if no datasets found
+        if len(dataset_row) == 0:
+            sleep(5)
+            self._query_fileview()
```

Comment:
why do we need to re-query if no datasets found?

Reply:
Before, it was re-querying every time the method was called. Now it will only re-query if no datasets were found as a last attempt to not error out.

Reply:
@milen-sage lingling mentioned that you might've wanted this to be removed? Is that the case?

Reply:
We can set expectations that if datasets were just added they may not immediately appear in schematic and DCA. That said, as @GiaJordan noted, this code now only retries if there are no datasets found (e.g. datasets may have been added for a first time in a project and may take a few seconds to be indexed). In the latter case the extra 5s is not too bad. As long as we are not adding the 5s overhead for every query, we can leave it in.

Reply:
@milen-sage cool, thank you for clarifying